src/OpenConext/ProfileBundle/Storage/SingleCookieStorage.php

```diff
@@ -42,13 +43,43 @@ class SingleCookieStorage implements EventSubscriberInterface
      */
     private $cookieValue;
 
-    public function __construct($cookieDomain, $cookieKey)
-    {
+    /**
+     * @var DateTime|0
+     */
+    private $cookieExpirationDate;
+
+    /**
+     * @var boolean
+     */
+    private $cookieSecure;
+
+    /**
+     * @var boolean
+     */
+    private $cookieHttpOnly;
+
+    public function __construct(
+        $cookieDomain,
+        $cookieKey,
+        DateTime $cookieExpirationDate = null,
```

Comment:
You can use `DateTimeInterface` here.

While writing this comment, I found that the only place where the supported PHP versions are "documented" is in the Travis build file. Profile should require specific PHP versions in its Composer file to prevent installing incompatible packages (for example PHP 7-only packages). The README should also state that only PHP 5.6 is officially supported. PHP 7 is only tested for forward compatibility. Would you create an issue for that? :wink: